eval/chat_benchmarks/MATH500/eval_instruct.py

```diff
@@ -60,21 +59,31 @@ def generate_responses(self, model: LM) -> Dict[str, Any]:
         all_instances = []
         if isinstance(model, lm_eval.models.huggingface.HFLM):
             model_name = model.pretrained
+        elif isinstance(model, lm_eval.models.openai_completions.OpenAIChatCompletion):
+            model_name = str(f"openai/{model.model}")
         else:
             model_name = model.model_args["model"]
-        system_prompt = SYSTEM_PROMPT[model_name]
         for idx, example in enumerate(examples):
             messages = [
-                {"role": "system", "content": system_prompt},
                 {"role": "user", "content": PROMPT.format(problem=example["problem"])},
             ]
-            templated_messages = model.apply_chat_template(messages)
 
             generation_args = {
                 "do_sample": False,
-                "max_gen_toks" if isinstance(model, VLLM) else "max_new_tokens": self.max_new_tokens,
                 "temperature": 0.7,
             }
+            # Add max tokens except for OpenAI models
+            if not isinstance(model, lm_eval.models.openai_completions.OpenAIChatCompletion):
+                generation_args["max_gen_toks" if isinstance(model, VLLM) else "max_new_tokens"] = self.max_new_tokens
+            else:
+                if "o1-mini" in model_name:  # o1-mini is a special case for OpenAI models
+                    generation_args["max_tokens"] = 32768
```

Comment:
since we are repeating the same code a bunch, maybe we should put this in a better place like self.compute?

Reply:
Sure. I need to actually do this for all benchmarks. let's merge this and i submit a follow up to unify across all benchmarks